model/transaction/event_test.go

```diff
@@ -143,8 +152,174 @@ func TestEventTransform(t *testing.T) {
+	}
+}
+
+func TestPayloadTransform(t *testing.T) {
+	hostname := "a.b.c"
+	architecture := "darwin"
+	platform := "x64"
+	timestamp := time.Now()
+
+	service := metadata.Service{Name: "myservice"}
+	system := &metadata.System{
+		Hostname:     &hostname,
+		Architecture: &architecture,
+		Platform:     &platform,
+	}
+
+	txValid := Event{Timestamp: timestamp}
+	txValidEs := common.MapStr{
+		"context": common.MapStr{
+			"service": common.MapStr{
+				"name":  "myservice",
+				"agent": common.MapStr{"name": "", "version": ""},
+			},
+		},
+		"processor": common.MapStr{
+			"event": "transaction",
+			"name":  "transaction",
+		},
+		"transaction": common.MapStr{
+			"duration": common.MapStr{"us": 0},
+			"id":       "",
+			"type":     "",
+			"sampled":  true,
+		},
+	}
+
+	txValidWithSystem := common.MapStr{
+		"processor": common.MapStr{
+			"event": "transaction",
+			"name":  "transaction",
+		},
+		"transaction": common.MapStr{
+			"duration": common.MapStr{"us": 0},
+			"id":       "",
+			"type":     "",
+			"sampled":  true,
+		},
+		"context": common.MapStr{
+			"system": common.MapStr{
+				"hostname":     hostname,
+				"architecture": architecture,
+				"platform":     platform,
+			},
+			"service": common.MapStr{
+				"name":  "myservice",
+				"agent": common.MapStr{"name": "", "version": ""},
+			},
+		},
+	}
+	txWithContext := Event{Timestamp: timestamp, Context: common.MapStr{"foo": "bar", "user": common.MapStr{"id": "55"}}}
+	txWithContextEs := common.MapStr{
+		"processor": common.MapStr{
+			"event": "transaction",
+			"name":  "transaction",
+		},
+		"transaction": common.MapStr{
+			"duration": common.MapStr{"us": 0},
+			"id":       "",
+			"type":     "",
+			"sampled":  true,
+		},
+		"context": common.MapStr{
+			"foo": "bar", "user": common.MapStr{"id": "55"},
+			"service": common.MapStr{
+				"name":  "myservice",
+				"agent": common.MapStr{"name": "", "version": ""},
+			},
+			"system": common.MapStr{
+				"hostname":     "a.b.c",
+				"architecture": "darwin",
+				"platform":     "x64",
+			},
+		},
+	}
+	spans := []*span.Span{{}}
+	txValidWithSpan := Event{Timestamp: timestamp, Spans: spans}
+	spanEs := common.MapStr{
+		"context": common.MapStr{
+			"service": common.MapStr{
+				"name":  "myservice",
+				"agent": common.MapStr{"name": "", "version": ""},
+			},
+		},
+		"processor": common.MapStr{
+			"event": "span",
+			"name":  "transaction",
+		},
+		"span": common.MapStr{
+			"duration": common.MapStr{"us": 0},
+			"name":     "",
+			"start":    common.MapStr{"us": 0},
+			"type":     "",
+		},
+		"transaction": common.MapStr{"id": ""},
+	}
+
+	tests := []struct {
+		Metadata *metadata.Metadata
+		Events   []transform.Transformable
+		Output   []common.MapStr
+		Msg      string
+	}{
+		{
+			Metadata: metadata.NewMetadata(nil, nil, nil, nil),
+			Events:   []transform.Transformable{},
```

Comment:
same note as above, iterating through `[] transform.Transformable` is moved to the `beater` pkg, you should be good by testing `Transform` on single events.